packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp

```diff
@@ -233,52 +236,58 @@ int executeInsertGlobalIndicesFESP_(const Teuchos::RCP<const Teuchos::Comm<int>
   // - sumIntoGlobalValues( 7,  [  2  3  7  6  ],  [  0  -1  2  -1  ])
   // - sumIntoGlobalValues( 6,  [  2  3  7  6  ],  [  -1  0  -1  2  ])
-  RCP<TimeMonitor> timerElementLoopMatrix = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("3) ElementLoop  (Matrix)")));
 
-  RCP<fe_matrix_t> fe_matrix = rcp(new fe_matrix_t(fe_graph));
-  RCP<fe_multivector_t> rhs =
-    rcp (new fe_multivector_t(domain_map, fe_graph->getImporter(), 1));
+  RCP<fe_matrix_type> fe_matrix;
+  RCP<fe_multivector_type> rhs;
 
-  scalar_2d_array_t element_matrix;
-  Kokkos::resize(element_matrix, 4);
-  Teuchos::Array<Scalar> element_rhs(4);
+  {
+    TimeMonitor timerElementLoopMatrix
+      (*TimeMonitor::getNewTimer ("3) ElementLoop  (Matrix)"));
 
-  Teuchos::Array<global_ordinal_t> column_global_ids(4);     // global column ids list
-  Teuchos::Array<Scalar> column_scalar_values(4);         // scalar values for each column
+    fe_matrix = rcp(new fe_matrix_type(fe_graph));
+    rhs = rcp (new fe_multivector_type(domain_map, fe_graph->getImporter(), 1));
 
-  // Loop over elements
-  Tpetra::beginFill(*fe_matrix,*rhs);
-  for(size_t element_gidx=0; element_gidx<mesh.getNumOwnedElements(); element_gidx++)
-  {
-    // Get the contributions for the current element
-    ReferenceQuad4(element_matrix);
-    ReferenceQuad4RHS(element_rhs);
+    Kokkos::View<local_ordinal_type[4][4], execution_space> element_matrix ("element_matrix");
+    Teuchos::Array<Scalar> element_rhs(4);
 
-    // Fill the global column ids array for this element
-    for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
-    {
-      column_global_ids[element_node_idx] = owned_element_to_node_ids(element_gidx, element_node_idx);
-    }
+    Teuchos::Array<global_ordinal_type> column_global_ids(4);     // global column ids list
+    Teuchos::Array<Scalar> column_scalar_values(4);         // scalar values for each column
 
-    // For each node (row) on the current element:
-    // - populate the values array
-    // - add the values to the fe_matrix.
-    // Note: hardcoded 4 here because we're using quads.
-    for(size_t element_node_idx=0; element_node_idx<4; element_node_idx++)
-    {
-      global_ordinal_t global_row_id = owned_element_to_node_ids(element_gidx, element_node_idx);
+    // Loop over elements
+    Tpetra::beginFill(*fe_matrix,*rhs);
+    for (size_t element_gidx = 0;
+         element_gidx < mesh.getNumOwnedElements ();
+         ++element_gidx) {
+      // Get the contributions for the current element
+      ReferenceQuad4(element_matrix);
+      ReferenceQuad4RHS(element_rhs);
 
-      for(size_t col_idx=0; col_idx<4; col_idx++)
-      {
-        column_scalar_values[col_idx] = element_matrix(element_node_idx, col_idx);
+      // Fill the global column ids array for this element
+      for (size_t element_node_idx=0;
+           element_node_idx < owned_element_to_node_ids.extent(1);
+           ++element_node_idx) {
```

Comment:
You know, modern monitors support more than 80 columns... :tongue:

Reply:
@csiefer2 Seriously though, if you're visually impaired and have to blow up the font, long lines are super annoying

Reply:
but thanks for the review :-D

Reply:
Point taken.  Just had to share the love, though :stuck_out_tongue:

Reply:
Clearly @mhoemmen  has seen the font on my laptop (probably from all the way across the big conference room).  Thanks!